include/nuttx/sched_note.h

```diff
@@ -105,16 +105,32 @@
   memset((s), 0, sizeof(struct note_filter_irq_s))
 #endif
 
-/* Note dump module tag definitions */
-
 #ifdef CONFIG_SCHED_INSTRUMENTATION_DUMP
-#  define NOTE_MODULE(a, b, c, d)  \
-  ((uint32_t)((a) & 0xff)        | \
-  ((uint32_t)((b) & 0xff) << 8)  | \
-  ((uint32_t)((c) & 0xff) << 16) | \
-  ((uint32_t)((d) & 0xff) << 24))
+#  define SCHED_NOTE_IP \
+          ({__label__ sched_note_here; sched_note_here: (uintptr_t)&&sched_note_here;})
```

Comment:
This seems dose not to work, __label__ can ensure the uniqueness of the current label, it can be guaranteed that the definition will not be duplicate in the code scope:
`: error: duplicate label ‘sched_note_here’`
this practice is very common, you can see that the linux kernel uses this extension a lot
https://github.com/torvalds/linux/blob/master/include/linux/instruction_pointer.h#L6

Reply:
Improved again, seems works fine.
```
#  define _SCHED_NOTE_PASTE(x, y) x ## y
#  define SCHED_NOTE_PASTE(x, y) _SCHED_NOTE_PASTE(x, y)
#  define SCHED_NOTE_LABEL \
          SCHED_NOTE_PASTE(sched_note_here, __LINE__)
#  define SCHED_NOTE_IP \
          ({SCHED_NOTE_LABEL: (uintptr_t)&&SCHED_NOTE_LABEL;})

```